### ISSUE TITLE
Aria-roledescription should not be a global attribute.

### ISSUE THREAD

**Issue by accdc** [w3c/aria#1132]
I've been getting reports that aria-roledescription is being misinterpreted as a description for a role, and not what the intended purpose for this attribute is. As a result, code like the following is going into production environments.

```
<label>Choose a username:
<input type="text" aria-roledescription="No special characters allowed" />
</label>
```

And

```
<button aria-roledescription="Set focus to hear description" aria-describedby="dynamicTooltipId" >Format</button>
```

If there are no limitations put on aria-roledescription within the spec, then this will eventually cause critical accessibility issues all across the web.

Unfortunately it is going to have to be the browsers and ATs that will need to restrict this. Authors will just use it for this purpose, regardless what the spec says.

**Comment by stes-acc** [comment]
I highly second a binding and more formal grammar proposal to be used for aria-roledescription string validators can use and warn if violations occur.

For instance, custom role names (and THIS is the intended use case) should be limited to two words maximum, some keywords may have to be used etc. We need clever ideas here.

This is also groundwork for ARIA 2.0 custom role definitions where role name is essential.

ARIA APG using aria-roledescription should be enhanced here and ARIA 1.x should contain respective mandatory regulations.

We must avoid that this VALUABLE property get corrupted by misuse and in the end the acc. web gurus spread a "do not use at all because confusing" message.

**Comment by LaurenceRLewis** [comment]
For aria-roledescription would it be worth adding a ‘note ’ similar to WCAG Success Criterion 2.5.3 Label in Name that recommends ‘A best practice is to have the text of the label at the start of the name.’

Except in the case if aria-roledescription the elements role is ‘at the start’. 

Taking @accdc examples:

`<label>Choose a username:
<input type="text" aria-roledescription="input type text No special characters allowed" />
</label>`

`<button aria-roledescription="Button Set focus to hear description" aria-describedby="dynamicTooltipId" >Format</button>`

**Comment by jnurthen** [comment]
@LaurenceRLewis - No - we don't want people to add hint text like this in roledescription at all. 
For non-interactive elements indeed I would like to see a SHOULD NOT use the role as part of the roledescription.
For many interactive widgets though it may be useful to encourage the widget role to be part of the text of the roledescripton but I don't think this should be a normative MUST. For example, if used on application we wouldn't want the text application to be used. I have a hard time seeing the use case on many widget types (like button) though as normally changing the accessible name would be better than changing the roledescription.
See the examples in the spec for `aria-roledescription="slide"` and in APG for `aria-roledescription="carousel"`
We should probably add an example for application in the future as that is the best use case I can think of for widgets (unless someone can come up with a better one)

**Comment by LaurenceRLewis** [comment]
Thanks @jnurthen As I now understand it to add a aria-roledescription to say a button element is problematic, for example the `<button>` is translated into different languages, whereas `<button aria-roledescription="This button does something">`  is not and the role description overides the semantic information.

Or do I get another down pointing thumb :-)

**Comment by carmacleod** [comment]
Probably should allow aria-roledescription on generic role.

**Comment by carmacleod** [comment]
APG should have guidance on using aria-roledescription.

Consider changing name to aria-rolealias or aria-rolereplacement some day.

**Comment by stes-acc** [comment]
> @LaurenceRLewis - No - we don't want people to add hint text like this in roledescription at all.
> For non-interactive elements indeed I would like to see a SHOULD NOT use the role as part of the roledescription.
> For many interactive widgets though it may be useful to encourage the widget role to be part of the text of the roledescripton but I don't think this should be a normative MUST. For example, if used on application we wouldn't want the text application to be used. I have a hard time seeing the use case on many widget types (like button) though as normally changing the accessible name would be better than changing the roledescription.
> See the examples in the spec for `aria-roledescription="slide"` and in APG for `aria-roledescription="carousel"`
> We should probably add an example for application in the future as that is the best use case I can think of for widgets (unless someone can come up with a better one)

There is much to add here.

I think a rule of thumb can be extracted from the wording in the  existing ARIA Spec : **If the base role is vague, you are allowed to totally override base role. If base role is an active element, string representing  base role should be appended to  to inform about potential keyboard usage.**

Unfortunately, the group has not decided on this very definition yet but I see it as just reasonable. Again, the background of this hesitation is that the requirements for screen readers how to speak roledescription **are currently under discussion (roledescription OR base role vs. roledescription AND base role)**. Personally I  think the override should be configurable in SR itself with a default value = OFF (!) but that’s just me. 

**Anyway, we should have group consensus what screen readers SHOULD do here first.**

**Comment by LaurenceRLewis** [comment]
Aside note: @stes-acc ’  but I don't think this should be a normative MUST’. I was not suggesting this is a MUST in the normative text. 

It will be beneficial when this role is fully documented. To date all use cases I have seen cause accessibility issues due to misuse. 

I would at some point like to understand how aria-roledescription should be used, which elements it CANNOT be used on, as well as guidelines as you have described above. 

The organisation I work for have just become a W3C member so I am hoping to become more involved with the ARIA WG in the future. Learning from you all. 😀

**Comment by cookiecrook** [comment]
Other than disallowing `aria-roledescription` on `generic`, I don't support removing it as a global. I recall seeing valid examples on elements not yet specified in ARIA, including some tags of SVG, and emerging HTML features prior to ARIA parity.

**Comment by stes-acc** [comment]
Results of deep dive on the topic:

1.	It is helpful  to mention base role in roledescription when using roledescription on widgets. The ARIA spec https://w3c.github.io/aria/#aria-roledescription chapter can be improved with a sentence explaining this.

2.	Any text not mentioning base role when using roledescription on widgets is problematic since it masks base role and is therefore considered as authoring error. Bad practices get no special mentioning neither in the ARIA specification nor the ARIA APG.

3.	It is expected that for overrides of base role of region containers etc. using roledescription on widgets, the roledescription is spoken instead the base role when focusable descendants are focused. Observed violations can be filed to the AT vendor and are expected to be acknowledged and fixed.

4.	The reason for overriding base role speech output with roledescription content was decided on group consensus when the property has been introduced to AT as result of feedback of blind users that were confused by double-speak of roledescription and base role. 

5.	The group has created a respective special issue [https://github.com/w3c/aria/issues/1480)] where all further discussion should be continued.

6.	Update with this info:

[18:08] <StefanS_> [https://github.com/w3c/aria/issues/1373]
[18:08] <StefanS_> [https://github.com/w3c/aria/issues/1132]
[18:09] <StefanS_> [https://github.com/w3c/aria/issues/1372]
